internal/core/services/workers_manager/workers_manager.go

```diff
@@ -0,0 +1,159 @@
+package workers_manager
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/topfreegames/maestro/internal/config"
+	"github.com/topfreegames/maestro/internal/core/entities"
+	"github.com/topfreegames/maestro/internal/core/ports"
+	"github.com/topfreegames/maestro/internal/core/services/operation_manager"
+	"github.com/topfreegames/maestro/internal/core/workers"
+	"go.uber.org/zap"
+)
+
+// configurations paths for the worker
+const (
+	// Sync period: waiting time window respected by workers in
+	// order to control executions
+	syncOperationWorkersIntervalPath = "check.operation.workers.interval"
+)
+
+// Default struct of WorkersManager service
+type WorkersManager struct {
+	configs                      config.Config
+	schedulerStorage             ports.SchedulerStorage
+	operationManager             operation_manager.OperationManager
+	CurrentWorkers               map[string]workers.Worker
+	runSyncOperationWorkers      bool
+	syncOperationWorkersInterval int
+}
+
+// Default constructor of WorkersManager
+func NewWorkersManager(configs config.Config, schedulerStorage ports.SchedulerStorage, operationManager operation_manager.OperationManager) *WorkersManager {
+	return &WorkersManager{
+		configs:                      configs,
+		schedulerStorage:             schedulerStorage,
+		operationManager:             operationManager,
+		CurrentWorkers:               map[string]workers.Worker{},
+		syncOperationWorkersInterval: configs.GetInt(syncOperationWorkersIntervalPath),
+	}
+}
+
+// Function to run a first sync and start a periodically sync worker
+func (w *WorkersManager) Start(ctx context.Context) error {
+
+	err := w.SyncOperationWorkers(ctx)
+	if err != nil {
+		zap.L().Error("initial sync operation workers failed", zap.Error(err))
```

Comment:
If we have failed to sync the first time, should we keep the process going?

Reply:
Returning the error